pkg/vcs/git.go

```diff
@@ -575,3 +575,17 @@ func (git *git) IsRelease(commit string) (bool, error) {
 	}
 	return len(tags1) != len(tags2), nil
 }
+
+func (git *git) cherryPickIfMissing(commit string) error {
```

Comment:
There's a hidden assumption that the commit-to-be-cherry-picked is either a parent of HEAD or does not exist. This is true for your Linux-based use case, but the code in this file is supposed to be generic, so this should ideally also cover other possible situations.

I'd suggest that you introduce e.g. a `git.cherryPickChanges` function instead and invoke it and `git.Contains` directly from `linux.go`.

Reply:
> There's a hidden assumption that the commit-to-be-cherry-picked is either a parent of HEAD or does not exist.

True, my assumption is the parent of HEAD during a bisection into past history

> this should ideally also cover other possible situations.

I assumed this to be a useful feature for all git kernels. The problem is once an issue is in the main branch of any git repo you need to work around it during bisection, as history rewriting is not viable.

> I'd suggest that you introduce e.g. a git.cherryPickChanges function instead and invoke it and git.Contains directly from linux.go.

That function would only consist of `return git.git("cherry-pick", "--no-commit", commit)`. As we currently only have a single call to this function in linux anyhow, we could also just move the code there for now and reconsider should we need more cherry picks in the future. Would that work for you?

Reply:
> As we currently only have a single call to this function in linux anyhow, we could also just move the code there for now and reconsider should we need more cherry picks in the future. Would that work for you?

Sounds good.

Reply:
Done